File: project-euler/problem 2/solution.ts
let sum_of_evens: number = 0;
let fibonacci_sequence: number[] = [1, 2];

// Create the Fibonacci sequence for numbers up to 4,000,000
while (Math.max(...fibonacci_sequence) <= 4000000) {
    let next_value: number = fibonacci_sequence[fibonacci_sequence.length -1] + fibonacci_sequence[fibonacci_sequence.length -2];

    if (next_value <= 4000000) {
        fibonacci_sequence.push(next_value);
    } else {
        break;
    }
}

// Sum the even numbers in the Fibonacci sequence from above
for (let index: number = 0; index < fibonacci_sequence.length; index++) {
    const element: number = fibonacci_sequence[index];

    if ((element % 2) == 0) {
        sum_of_evens += element;
    }
}

console.log(sum_of_evens);

File: project-euler/problem 2/solution.dart
void main() {
  // All Fibonacci numbers
  List<int> fibonacciNumbers = [1, 2];
  // Even Fibonacci numbers
  List<int> evenFibonacciNumbers = [];

  // Find the Fibonacci numbers equal to or less than 4,000,000
  while ((fibonacciNumbers[fibonacciNumbers.length - 1] +
          fibonacciNumbers[fibonacciNumbers.length - 2]) <=
      4000000) {
    // Add the new Fibonacci number to the existing sequence
    fibonacciNumbers.add(fibonacciNumbers[fibonacciNumbers.length - 1] +
        fibonacciNumbers[fibonacciNumbers.length - 2]);
  }

  // Filter out the odd Fibonacci numbers
  for (var i = 0; i < fibonacciNumbers.length; i++) {
    if (fibonacciNumbers[i].remainder(2) == 0) {
      evenFibonacciNumbers.add(fibonacciNumbers[i]);
    }
  }

  print(evenFibonacciNumbers.reduce((a, b) => a + b));
}
